src/modules/storage/LedgerStorage.ts

```diff
@@ -153,12 +153,15 @@ export class LedgerStorage extends Storages
                 {
                     let block: Block = new Block();
                     block.parseJSON(data);
+                    await this.begin();
                     await saveBlock(this, block);
                     await this.putTransactions(block);
                     await this.putEnrollments(block);
+                    await this.commit();
```

Comment:
It seems very backward to have an await per line here. Those operations are dependent on one another, so they should be in the same promise.

Reply:
Yes, that's right.
I think it can do a common, dependent task within Begin.